### PR TITLE
[libe57] Add new port

### PR DESCRIPTION
**Pull request by maz-1**
**Describe the pull request**

- #### What does your PR fix?  
  Add support for libe57

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
  Every officially supported triplets builds without errors.
  No.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/libe57/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,52 @@
+include(vcpkg_common_functions)
```
Comment (JonLiu1993) [inline ports/libe57/portfile.cmake:1]: @maz-1 ,Thanks for your pr,
`include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/libe57/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "libe57",
+  "version-semver": "1.1.312",
+  "port-version": 0,
```
Comment (JonLiu1993) [inline ports/libe57/vcpkg.json:4]: We should delete "port-version"

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libe57/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+file(INSTALL ${SOURCE_PATH}/include/LASReader.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57)
+file(INSTALL ${SOURCE_PATH}/include/time_conversion/time_conversion.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57/time_conversion)
+file(INSTALL ${SOURCE_PATH}/include/time_conversion/basictypes.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57/time_conversion)
+file(INSTALL ${SOURCE_PATH}/include/time_conversion/constants.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57/time_conversion)
+file(INSTALL ${SOURCE_PATH}/include/time_conversion/gnss_error.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57/time_conversion)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/bin)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin)
```
Comment (JonLiu1993) [inline ports/libe57/portfile.cmake:44]: @maz-1, can you clarify why these directories were deleted

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libe57/portfile.cmake (hunk 3 above)
Comment (maz-1) [reply ports/libe57/portfile.cmake:44]: executables, they are already installed to the corrent directory via vcpkg_copy_tools

Comment (maz-1) [reply ports/libe57/portfile.cmake]: I created this port last year when vcpkg_copy_tools doesnt exist yet, I removed them now.

**Comment by JonLiu1993** [comment]
@maz-1 ,Thanks for your pr.
If this pr has been modified, please click "ready for review" to let me know

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@maz-1 ,There are too many modifications in this patch. Can you open an issue in upstream to let Upstream know?

File: ports/libe57/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,51 @@
+set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/E57RefImpl_src-1.1.312)
+vcpkg_download_distfile(ARCHIVE
```
Comment (JonLiu1993) [inline ports/libe57/portfile.cmake:2]: We have a function specifically downloaded from sourcefor, we recommend using _vcpkg_from_sourceforge_

File: ports/libe57/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+vcpkg_extract_source_archive_ex(
+  OUT_SOURCE_PATH SOURCE_PATH
+  ARCHIVE ${ARCHIVE}
+  PATCHES 
+    "0001_cmake.patch"
+    "0002_replace_tr1_with_cpp11.patch"
+    "0003_fix_osx_support.patch"
+)
+
+vcpkg_configure_cmake(
```
Comment (JonLiu1993) [inline ports/libe57/portfile.cmake:17]: Please use the new function; vcpkg_cmake_configure()

File: ports/libe57/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+)
+
+file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/debug/share/libe57)
+
+vcpkg_install_cmake()
```
Comment (JonLiu1993) [inline ports/libe57/portfile.cmake:24]: Please use the new function; vcpkg_cmake_install()

File: ports/libe57/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+vcpkg_fixup_cmake_targets()
```
Comment (JonLiu1993) [inline ports/libe57/portfile.cmake:25]: Please use the new function: vcpkg_cmake_config_fixup()

File: ports/libe57/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    
+file(INSTALL ${SOURCE_PATH}/README.TXT DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL ${SOURCE_PATH}/src/refimpl/E57RefImplConfig.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/e57refimpl/)
+
+file(INSTALL ${SOURCE_PATH}/include/E57Simple.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57)
+file(INSTALL ${SOURCE_PATH}/include/LASReader.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57)
+file(INSTALL ${SOURCE_PATH}/include/time_conversion/time_conversion.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57/time_conversion)
+file(INSTALL ${SOURCE_PATH}/include/time_conversion/basictypes.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57/time_conversion)
+file(INSTALL ${SOURCE_PATH}/include/time_conversion/constants.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57/time_conversion)
+file(INSTALL ${SOURCE_PATH}/include/time_conversion/gnss_error.h DESTINATION ${CURRENT_PACKAGES_DIR}/include/e57/time_conversion)
```
Comment (JonLiu1993) [inline ports/libe57/portfile.cmake:41]: Can this part be implemented in CMakeLists?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libe57/portfile.cmake (hunk 5 above)
Comment (maz-1) [reply ports/libe57/portfile.cmake:17]: > @maz-1 ,There are too many modifications in those patches. Can you open an issue in upstream to let Upstream know?

I opened a issue in their sourceforge bug reports, although it seems their bug reports are not actively maintained.

**Comment by maz-1** [comment]
> @maz-1 ,There are too many modifications in those patches. Can you open an issue in upstream to let Upstream know?

I opened a issue in their sourceforge bug reports, although it seems their bug reports are not actively maintained.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@maz-1 ,Thanks for your new port!

**Comment by BillyONeal** [comment]
Thanks for your help!

**Pull request merged**